services/src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -43,71 +43,89 @@
 import java.util.concurrent.atomic.AtomicLong;
 
 
-/**
- * This sets up the producers used by Kafka Monitoring and some of the reported metrics.
- */
 public class ProduceService implements Service {
   private static final Logger LOG = LoggerFactory.getLogger(ProduceService.class);
   private static final String METRIC_GROUP_NAME = "produce-service";
 
   private final String _name;
   private final ProduceMetrics _sensors;
-  private final KMBaseProducer _producer;
-  private final ScheduledExecutorService _executor;
+  private KMBaseProducer _producer;
+  private ScheduledExecutorService _executor;
+  private final ScheduledExecutorService _checkAddPartitionsExecutor;
```

Comment:
Sorry if I am paranoid about the name here -- would it be better to name it handleNewPartitionsExecutor, given that the runnable is named HandleNewPartitions?